File: java-context-os/src/main/java/com/owencli/contextos/pipeline/middleware/IntentMiddleware.java
package com.owencli.contextos.pipeline.middleware;

import com.owencli.contextos.pipeline.PipelineContext;
import com.owencli.contextos.pipeline.PipelineMiddleware;
import com.owencli.contextos.intent.TaskParser;
import java.util.concurrent.CompletableFuture;

public class IntentMiddleware implements PipelineMiddleware {
    private final TaskParser tp;
    public IntentMiddleware(TaskParser t) { this.tp = t; }
    public String name() { return "intent"; }
    public int order() { return 100; }
    public CompletableFuture<Void> execute(PipelineContext ctx) {
        return tp.parse(ctx.userInput()).thenAccept(ctx::setTaskSpec);
    }
}

File: context_os/pipeline/middlewares/intent_middleware.py
"""IntentMiddleware — 意图理解。"""

from context_os.intent.parser import TaskParser
from context_os.pipeline.context import PipelineContext
from context_os.pipeline.middleware import PipelineMiddleware


class IntentMiddleware(PipelineMiddleware):
    """解析用户输入 → TaskSpec。"""

    def __init__(self, task_parser: TaskParser):
        self._tp = task_parser

    def name(self) -> str:
        return "intent"

    def order(self) -> int:
        return 100

    async def execute(self, ctx: PipelineContext) -> None:
        ctx.task_spec = await self._tp.parse(ctx.user_input)
